torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -438,6 +438,25 @@ void Reducer::mark_variable_ready_sparse(size_t variable_index) {
         logger_,
         "Expected variable to have sparse gradient.");
 
+    // Copy the indices of sparse metadata
+    if (sparse_metadata_) {
+      grad = grad.coalesce();
+      REDUCER_CHECK(
```

Comment:
since we're turning this on by default now, the check can probably be moved to the Reducer constructor. Here, it would probably be called on every parameter every iteration.